src/components/discreteSlider/index.tsx

```diff
@@ -71,30 +63,35 @@ const DiscreteSlider = <T,>({
   };
 
   return (
-    <ChakraSlider
+    <Slider.Root
       step={1}
       max={sliderMarks.length - 1 + getItemOffset(applyIndentation)}
-      value={sliderStepValue}
-      onChange={handleOnChange}
-      focusThumbOnChange={false}
+      value={[sliderStepValue]}
+      onValueChange={(details) =>
+        handleOnChange(details.value[0] ?? sliderStepValue)
+      }
     >
-      {sliderMarks?.map(({ stepValue, label }, index) => {
-        return (
-          <ChakraSliderMark
-            key={index}
-            value={stepValue as number}
-            fontWeight={sliderStepValue === stepValue ? 'bold' : 'normal'}
-          >
-            <Box style={{ pointerEvents: 'all' }}>{label}</Box>
-          </ChakraSliderMark>
-        );
-      })}
-      <SliderTrack>
-        <SliderFilledTrack />
-      </SliderTrack>
-      <SliderThumb />
-    </ChakraSlider>
+      <Slider.Control>
+        <Slider.Track>
+          <Slider.Range />
+        </Slider.Track>
+        <Slider.Thumbs />
+        <Slider.MarkerGroup>
+          {sliderMarks?.map(({ stepValue, label }, index) => (
```

Comment:
Aren't marks required:
```suggestion
          {sliderMarks.map(({ stepValue, label }, index) => (
```